botocore/client.py

```diff
@@ -202,6 +207,7 @@ def _get_client_args(self, service_model, region_name, is_secure,
         # Add any additional s3 configuration for client
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
+        self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
```

Comment:
I was concerned that this may be unregistering the non-copied version of the event emitter, but you are fine. Ideally this would be moved to the BaseClient class where we handle the other s3 registering and unregistering, but we never have access to the customer provided endpoint url so that is not doable without a big refactoring. So I think I am fine with this here.